src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -17,43 +17,23 @@ using namespace Microsoft::Console::Render;
 // Routine Description:
 // - Creates a CustomTextLayout object for calculating which glyphs should be placed and where
 // Arguments:
-// - factory - DirectWrite factory reference in case we need other DirectWrite objects for our layout
-// - analyzer - DirectWrite text analyzer from the factory that has been cached at a level above this layout (expensive to create)
-// - format - The DirectWrite format object representing the size and other text properties to be applied (by default) to a layout
-// - formatItalic - The italic variant of the format object representing the size and other text properties for italic text
-// - font - The DirectWrite font face to use while calculating layout (by default, will fallback if necessary)
-// - fontItalic - The italic variant of the font face to use while calculating layout for italic text
-// - width - The count of pixels available per column (the expected pixel width of every column)
-// - boxEffect - Box drawing scaling effects that are cached for the base font across layouts.
-CustomTextLayout::CustomTextLayout(gsl::not_null<IDWriteFactory1*> const factory,
-                                   gsl::not_null<IDWriteTextAnalyzer1*> const analyzer,
-                                   gsl::not_null<IDWriteTextFormat*> const format,
-                                   gsl::not_null<IDWriteTextFormat*> const formatItalic,
-                                   gsl::not_null<IDWriteFontFace1*> const font,
-                                   gsl::not_null<IDWriteFontFace1*> const fontItalic,
-                                   size_t const width,
-                                   IBoxDrawingEffect* const boxEffect) :
-    _factory{ factory.get() },
-    _analyzer{ analyzer.get() },
-    _format{ format.get() },
-    _formatItalic{ formatItalic.get() },
-    _formatInUse{ format.get() },
-    _font{ font.get() },
-    _fontItalic{ fontItalic.get() },
-    _fontInUse{ font.get() },
-    _boxDrawingEffect{ boxEffect },
+// - dxFontRenderData - The DirectWrite font render data for our layout
+CustomTextLayout::CustomTextLayout(gsl::not_null<DxFontRenderData*> const fontRenderData) :
+    _fontRenderData{ fontRenderData },
+    _formatInUse{ fontRenderData->DefaultTextFormat().Get() },
+    _fontInUse{ fontRenderData->DefaultFontFace().Get() },

```

Comment:
Oh my gosh that is so much cleaner. Though any reason to not also put the "in use" things inside the `DxFontRenderData` as well and let them return the pointer to the correct one on a Getter (with a setter to change which is which?)

Reply:
I thought about it. But I sorta want to make `DxFontRenderData` more "stateless" & purely about fonts itself, not about "how we use the fonts". The `UpdateFont` will be the only "setter" API. This should allow us to share one `DxFontRenderData` with more consumers. Also we canextend `DxFontRenderData` more easily without having to worry about any state from the outside. Sounds reasonable to u?

Reply:
That's a good justification. I accept. :)